### DISCUSSION TITLE
Storage facade fake not swapping for a performed request

### DISCUSSION THREAD

**Discussion by tkuijer** [laravel/framework#38630] [Q&A]
- Laravel Version: 8.58.0
- PHP Version: 8.0.10
- Database Driver & Version: MySQL

### Description:
When attempting to test file uploads in my application, the laravel documentation tells me to call [Storage::fake()](https://laravel.com/docs/8.x/http-tests#testing-file-uploads). This returns a FileSystemAdapter configured for the testing environment. But doesn't appear to be application-wide.

The fake file that gets uploaded isn't get stored in the testing path, but instead gets stored to the regular storage folder.

I attempted both the described [method ](https://laravel.com/docs/8.x/filesystem#file-uploads)for handling file uploads, and the referenced method using the Storage facade. Both result in the file getting stored in the wrong folder.

### Steps To Reproduce:
1) Copy Test from https://laravel.com/docs/8.x/http-tests#testing-file-uploads
2) Copy Controller logic from https://laravel.com/docs/8.x/filesystem#file-uploads
3) Create route in routes/web.php: `Route::post('/avatars', [UserAvatarController::class, 'update']);`

### Expected result:
- The faked uploaded file is stored in `storage/framework/testing/disks/avatars`
- Test passes

### Actual result:
- The faked uploaded file gets stored in `storage/app/avatars`
- Test fails

### Repository demonstrating the issue:
https://github.com/tkuijer/laravel-storagefake-bugreport

**Comment by Jubeki** [comment]
In the Avatars Controller you are storing the file in `avatars` path on disk `local`.

Instead you want to save it to the disk `avatars`:
```php
$path = $request->file('avatar')->store('', 'avatars');

// or 
$path = $request->file('avatar')->store('', [
    'disk' => 'avatars',
]);
```


Link to Documentation: https://laravel.com/docs/8.x/requests#storing-uploaded-files
